chromadb/utils/embedding_functions.py

```diff
@@ -334,13 +339,19 @@ def _init_model_and_tokenizer(self) -> None:
             # https://github.com/UKPLab/sentence-transformers/blob/3e1929fddef16df94f8bc6e3b10598a98f46e62d/docs/_static/html/models_en_sentence_embeddings.html#LL480
             self.tokenizer.enable_truncation(max_length=256)
             self.tokenizer.enable_padding(pad_id=0, pad_token="[PAD]", length=256)
+            if self._preferred_providers is None or len(self._preferred_providers) == 0:
```

Comment:
@HammadB , it occurs to me that there maybe a case where there's only  1 available provider (e.g. M2 ) where the warning can be suppressed and instead default to that one available silently (kind of the existing behaviour)